js/layout-javascript/agenda-code.js

```diff
@@ -1239,6 +1245,7 @@ DynamicList.prototype.renderLoopHTML = function () {
         _this.$container.find('.agenda-list-day-holder').eq(0).addClass('active');
       }
 
+      // Break render cycle if there is no more data
       if (!nextBatch.length) {
```

Comment:
Can you explain why this `if` block is still necessary?

Reply:
This block is necessary to render no results HTML in case there are no results.